command.go

```diff
@@ -850,42 +852,64 @@ func (c *Command) HasSubCommands() bool {
 	return len(c.commands) > 0
 }
 
+// Determine if a command is available as a non-help command (this includes all
```

Comment:
I said I was done commenting, but maybe I lied? Your comments are great. If you start them with the name of the following function they will get turned into godocs...   So this would start with `// IsAvailableCommand`